### ISSUE TITLE
Add support for unions of copy types

### ISSUE THREAD

**Issue by zommiommy** [zommiommy/mem_dbg-rs#11]
The problem of unions is that we shouldn't follow uninitialized pointers and references, but if we only have copy types with no references then `core::mem::size_of` is just the max size of the variants. So we can implement MemSize, while MemDbg should threat them as an opaque struct with no fields but the correct size.

**Comment by vigna** [comment]
Yes I was a bit worried about this PR.

**Comment by zommiommy** [comment]
Yes, I merged the union pr, but I kept only the impl for unions with only 1 variant as they are correct.

For multiple variants is much more subtle so I think it's best to do it in a second PR
